### DISCUSSION TITLE
Has anyone gotten a tool to work yet?

### DISCUSSION THREAD

**Discussion by Jonneal3** [mckaywrigley/chatbot-ui#1414] [Help]
HI guys - Ive done a deep dive on whats gong on in the back end here over the last week.  I cannot seem t, in any way get a tool request to work. Has anyone performed a tool request yet? If so, can you post what you did so I can replicate? Thanks! - Jon

**Comment by secondtruth** [comment]
I had problems with tools too. Turned out, the parser didn't like the `$ref`s in the OpenAPI schema. I inlined all `schema` references, then it worked.

**Reply by tranhasimple** [reply]
@secondtruth Can you explain your idea and give me an example for that?

**Reply by secondtruth** [reply]
@tranhasimple 

![Screenshot](https://github.com/user-attachments/assets/2d4b26db-bb3e-49a2-9551-765be751ecc8)

**Reply by tranhasimple** [reply]
So I do sth like that but it doesn't work. It looks like:
![image](https://github.com/user-attachments/assets/f64918af-b02b-4b74-a4c0-f7ed590a4452)

**Reply by DunaSpice** [reply]
@tranhasimple Where did this lead?

**Reply by crystal-coding-time** [reply]
Any updates or links to resources? I can't seem to find any posts on tools people are using and how they format them either through the UI or direct code updates.
